src/httpServer/server.ts

```diff
@@ -66,10 +66,26 @@ export class DBOSHttpServer {
   async listen(port: number) {
     try {
       await this.checkPortAvailability(port, "127.0.0.1");
+    } catch (error) {
+      const err = error as NodeJS.ErrnoException;
+      if (err.code === 'EADDRINUSE') {
+        this.logger.error(`Port ${port} is already used for IPv4 address "127.0.0.1". Please use the -p option to choose another port.\n${err.stack}`);
```

Comment:
Was the stack being gobbled? The `error()` handler is configured to add the stack itself if the input object is not an error type: https://github.com/dbos-inc/dbos-transact/blob/9cd4357a862998bc5db809228013de8cf0c10708/src/telemetry/logs.ts#L90

Reply:
This is an example output on my end:

```
2024-04-14 19:26:15 [error]: Port 3000 is already used for IPv6 address "::1". Please use the -p option to choose another port.
Error: listen EADDRINUSE: address already in use ::1:3000
    at Server.setupListenHandle [as _listen2] (node:net:1872:16)
    at listenInCluster (node:net:1920:12)
    at doListen (node:net:2075:7)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
    at GlobalLogger.error (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/telemetry/logs.js:63:65)
    at DBOSHttpServer.listen (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/httpServer/server.js:97:29)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DBOSRuntime.initAndStart (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/dbos-runtime/runtime.js:61:28)
    at async Command.<anonymous> (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/dbos-runtime/cli.js:28:5)
```

Reply:
Okay in my latest commit I changed to print `err.message` instead. Because additional stack doesn't seem to be very helpful.

Reply:
Example output now:
```
2024-04-14 20:37:32 [error]: Port 3000 is already used for IPv6 address "::1". Please use the -p option to choose another port.
listen EADDRINUSE: address already in use ::1:3000
    at GlobalLogger.error (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/telemetry/logs.js:63:65)
    at DBOSHttpServer.listen (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/httpServer/server.js:97:29)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DBOSRuntime.initAndStart (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/dbos-runtime/runtime.js:61:28)
    at async Command.<anonymous> (/Users/qianli/DBOS-projects/test-early-access/bigrelease/node_modules/@dbos-inc/dbos-sdk/dist/src/dbos-runtime/cli.js:28:5)
```